### PR TITLE
Reddit

### PR DESCRIPTION
**Pull request by awsomearvinder**

### CODE REVIEW

**Comment by awsomearvinder** [comment]
This branch adds support for parsing r/buildapcsales and updating a embed on the pcpartdeals channel. It does add a new field called "reddit_secret" in the config.json, I've sent a private message with the secret in DM's.

File: RomanPort.PCParadiseBot/Entities/IDiscordFilter.cs [hunk 1]
```diff
@@ -27,11 +25,12 @@ private async Task Discord_MessageCreated(MessageCreateEventArgs e)
                 //Check if supported
                 if (await IsAccepted(e))
                 {
-                    OnAccepted(e);
+                    await OnAccepted(e);
```
Comment (Roman-Port) [inline RomanPort.PCParadiseBot/Entities/IDiscordFilter.cs]: Awaiting this causes a lock while it's running.

File: RomanPort.PCParadiseBot/Program.cs [hunk 2]
```diff
@@ -14,7 +16,7 @@ class Program
     {
         public static DiscordClient discord;
         public static List<PCModule> modules;
-        
+        public static RedditClient reddit;
```
Comment (Roman-Port) [inline RomanPort.PCParadiseBot/Program.cs]: You should store this as a static variable inside of your own module, not in Program.cs

**Review by Roman-Port — CHANGES_REQUESTED** [review]

File: RomanPort.PCParadiseBot/Entities/IDiscordFilter.cs (hunk 1 above)
Comment (awsomearvinder) [reply RomanPort.PCParadiseBot/Entities/IDiscordFilter.cs]: Can we use await Task.Run() instead? that would run it in it's own background thread which would avoid the locking, and would get rid of the warnings due to it running synchronously - which may cause issue in the future.

File: RomanPort.PCParadiseBot/Program.cs (hunk 2 above)
Comment (awsomearvinder) [reply RomanPort.PCParadiseBot/Program.cs]: Fixed.

**Comment by awsomearvinder** [comment]
Force pushed in order due to me forgetting to add Program.cs before commiting the PartSales module - had to edit my local git history to add that in to previous commit.

File: RomanPort.PCParadiseBot/Entities/IDiscordFilter.cs (hunk 1 above)
Comment (awsomearvinder) [reply RomanPort.PCParadiseBot/Entities/IDiscordFilter.cs]: Added a commit to show what I'm talking about.

**Pull request merged**